### PR TITLE
Update data stream lifecycle telemetry to track global retention

### PR DESCRIPTION
**Pull request by gmarouli**
Currently, the data stream lifecycle telemetry has the following structure:
```
{
....
  "data_lifecycle" : {
    "available": true,
    "enabled": true,
    "count": 0,
    "default_rollover_used": true,
    "retention": {
        "minimum_millis": 0,
        "maximum_millis": 0,
        "average_millis": 0.0
    }
  }....
```

In the snippet above you can see that we track:

- The amount of data streams managed by the data stream lifecycle by `count`
- If the default rollover has been overwritten by `default_rollover_used`
- The min, max and average of the `data_retention` configured on a data stream level.

In this PR we propose the following extention:
```
....
  "data_lifecycle" : {
    "available": true,
    "enabled": true,
    "count": 0,
    "default_rollover_used": true,
    "effective_retention": { 
        "retained_data_streams": 5,
        "minimum_millis": 0, # Only if retained data streams > 1
        "maximum_millis": 0,
        "average_millis": 0.0
    },
    "data_retention": {
        "configured_data_streams": 5,
        "minimum_millis": 0, # Only if retained data streams > 1
        "maximum_millis": 0,
        "average_millis": 0.0
    },
    "global_retention": {
      "default": {
         "defined": true/false,
	  "affected_data_streams": 0,
         "millis": 0 
      },
      "max": {
         "defined": true/false,
	  "affected_data_streams": 0,
         "millis": 0 
      }
    }
```

With this extension we are tracking:

- The amount of data streams managed by the data stream lifecycle by `count`
- If the default rollover has been overwritten by `default_rollover_used`
- The min, max and average of the `data_retention` configured on a data stream level and the number of data streams that have it configured. We add the min, max and avg only if there are data streams with data retention configuration to avoid messing with the stats in a dashboard.
- The min, max and average of the `effective_retention` and the number of data streams that are retained. We add the min, max and avg only if there are retained data streams to avoid messing with the stats in a dashboard.
- Global retention stats, if they are defined, if the number of the affected data streams and the actual value.

The above metrics allow us to answer questions like:

- How many data streams are affected by global retention.
- How big is the difference between the longest data retention compared to max global retention.
- How much does the effective retention diverging from the data retention, this will show the impact of the global retention.

### CODE REVIEW

File: docs/reference/rest-api/usage.asciidoc [hunk 1]
```diff
@@ -377,10 +377,19 @@ GET /_xpack/usage
     "enabled": true,
     "count": 0,
     "default_rollover_used": true,
-    "retention": {
```
Comment (masseyke) [inline docs/reference/rest-api/usage.asciidoc:380]: This is a breaking change -- isn't that going to cause problems?

Comment (gmarouli) [reply docs/reference/rest-api/usage.asciidoc:380]: Considering that this is the usage endpoint that is not wildly used we have passed it from the committee and we got approval as long as it's not in a patch release.

File: docs/changelog/111972.yaml [hunk 2]
```diff
@@ -5,11 +5,13 @@ type: feature
...
+  body: |-
+    Data stream lifecycle now supports configuring retention on a cluster level,
+    namely global retention. Global retention \nallows us to configure two different
+    retentions:
+
+    - `data_streams.lifecycle.retention.default` is applied to all data streams managed
+    by the data stream lifecycle that do not have retention defined on the data stream level.
+    - `data_streams.lifecycle.retention.max` is applied to all data streams managed by the
+    data stream lifecycle and it allows any data stream \ndata to be deleted after the `max_retention` has passed.
   notable: true
```
Comment (gmarouli) [inline docs/changelog/111972.yaml:17]: This is not related to this change but I saw it looking a bit unreadable so I changed it to this format.

**Comment by gmarouli** [comment]
@elasticmachine update branch

**Comment by gmarouli** [comment]
@elasticmachine update branch

**Comment by gmarouli** [comment]
Tests muted in https://github.com/elastic/elasticsearch-serverless/issues/2726

**Review by masseyke — APPROVED** [review]
LGTM

**Comment by gmarouli** [comment]
@elasticmachine update branch

**Pull request merged**